agent/src/runtime_connectors/podman/podman_runtime_config.rs

```diff
@@ -14,6 +16,17 @@ pub struct PodmanRuntimeConfig {
     pub command_args: Vec<String>,
 }
 
+impl From<PodmanRuntimeConfig> for PodmanRunConfig {
```

Comment:
The PodmanRunConfig was introduced, because it feels unnatural that the PodmanRuntimeConfig is exposed from the podman folder to outside, so that podman_cli, which lies outside the podman folder, can use it. So you wanted to decouple the PodmanRuntimeConfig which lies in a sub component that is used by the parent podman_cli stuff? Do I understand this correctly?

I can see that all what podman_runtime does is a "try_from" from WorkloadSpec into PodmanRuntimeConfig and then it passes it to podman_run where it is converted again into a PodmanRunConfig.
For me that looks like that we shall use only the PodmanRunConfig or at least one of both types to get rid of the double conversions.

Maybe we can do the following: podman_cli is more low level then the runtime that uses it. So why not changing the folder structure so that podman_cli is low level and kube and podman runtime can use it. We introduce also a shared high level type for PodmanRuntimeConfig that is used by podman runtime and podman_cli. That podman_cli as a low level  component uses a high level data structure seems not a problem for me, but we save conversion code. Or you pass just the workload_spec to the podman_run and the podman_run does the conversion by itself.

What do you think?

Reply:
To the first paragraph, yes, you are correct, we don't want to leak out details in the wrong direction. 

As for the multiple conversions, one can view things in multiple ways:

PodmanRuntimeConfig is something clearly linked and dependent on the `podman` runtime. For this reason it shall be defined under the `podman` folder which contains all things related to the runtime connector for Podman.  
The podman_cli is a helper module that was targeted at bundling all Podman cli functionalities to one place. It shall have it's own interface s.t. it can be exchanged if needed. Out of pure coincidence 🤣, it luckily happens, that the `podman` runtime uses a config which has the same format as the one required by the podman_cli.